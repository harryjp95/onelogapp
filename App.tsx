import React from 'react'
import { StatusBar } from 'react-native'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'

import AccountScreen from './src/Components/Screens/Account'
import AnalyticsScreen from './src/Components/Screens/Analytics'
import HomeScreen from './src/Components/Screens/Home'
import NewReadingScreen from './src/Components/Screens/NewReading'
import SettingsScreen from './src/Components/Screens/Settings'

import MainHeader from './src/Components/MainHeader'

const Stack = createStackNavigator()

const App: React.FC = () => {
  return (
    <NavigationContainer>
      <StatusBar barStyle="dark-content" />
      <MainHeader />
      <Stack.Navigator initialRouteName="Home" screenOptions={{ headerShown: false, animationEnabled: false }}>
        <Stack.Screen name="Settings" component={SettingsScreen} />
        <Stack.Screen name="Account" component={AccountScreen} />
        <Stack.Screen name="Home" component={HomeScreen} />
        <Stack.Screen name="NewReading" component={NewReadingScreen} />
        <Stack.Screen name="Analytics" component={AnalyticsScreen} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default App
