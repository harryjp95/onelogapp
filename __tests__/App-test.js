import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../App'

it('renders correctly', () => {
  const app = renderer.create(<App />)

  expect(app).toMatchSnapshot()
})
