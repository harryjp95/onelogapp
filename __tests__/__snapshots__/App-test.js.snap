// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "backgroundColor": "black",
        "height": 48,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 24,
          "padding": 10,
          "textAlign": "center",
          "textDecorationLine": "underline",
        }
      }
    >
      OneLog
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#ebebeb",
        "flexDirection": "column",
        "height": 1214,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ebebeb",
          "flexDirection": "column",
          "height": 1214,
          "justifyContent": "flex-start",
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 5,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="black"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
        <BVLinearGradient
          colors={
            Array [
              4293651435,
              4286611584,
              4293651435,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "height": 0.5,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 5,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="black"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
        <BVLinearGradient
          colors={
            Array [
              4293651435,
              4286611584,
              4293651435,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "height": 0.5,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 5,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="black"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
        <BVLinearGradient
          colors={
            Array [
              4293651435,
              4286611584,
              4293651435,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "height": 0.5,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 5,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <ActivityIndicator
            animating={true}
            color="black"
            hidesWhenStopped={true}
            size="small"
          />
        </View>
        <BVLinearGradient
          colors={
            Array [
              4293651435,
              4286611584,
              4293651435,
            ]
          }
          endPoint={
            Object {
              "x": 1,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0,
              "y": 1,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "height": 0.5,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "flexDirection": "row",
        "height": 48,
        "justifyContent": "space-around",
        "textAlign": "center",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/Assets/Images/NavBarSettings.png",
          }
        }
        style={
          Object {
            "height": 30,
            "tintColor": "#ebebeb",
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/Assets/Images/NavBarAccount.png",
          }
        }
        style={
          Object {
            "height": 30,
            "tintColor": "#ebebeb",
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/Assets/Images/NavBarHome.png",
          }
        }
        style={
          Object {
            "height": 30,
            "tintColor": "#ebebeb",
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/Assets/Images/NavBarNewReading.png",
          }
        }
        style={
          Object {
            "height": 30,
            "tintColor": "#ebebeb",
            "width": 30,
          }
        }
      />
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/Assets/Images/NavBarAnalytics.png",
          }
        }
        style={
          Object {
            "height": 30,
            "tintColor": "#ebebeb",
            "width": 30,
          }
        }
      />
    </View>
  </View>
</RCTSafeAreaView>
`;
